File: binary_search_go/binary_search.go
package main

import "fmt"

func binarySearch(arr []int, num int) bool {
	first := 0
	last := len(arr) - 1
	found := false

	for first <= last && !found {
		mid := (first + last) / 2
		if arr[mid] == num {
			found = true
		} else {
			if num < arr[mid] {
				last = mid - 1
			} else {
				first = mid + 1
			}
		}
	}
	return found
}

func main() {
	arr := []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13}
	fmt.Println(binarySearch(arr, 11))
	fmt.Println(binarySearch(arr, 25))
	fmt.Println(binarySearch(arr, 2))
}

File: binary_search_typescript/binary_search.js
var binarySearch = function (arr, num) {
  var first = 0;
  var last = arr.length - 1;
  var found = false;
  while (first <= last && !found) {
    var mid = Math.floor((first + last) / 2);
    if (arr[mid] === num) {
      found = true;
    } else {
      if (num < arr[mid]) {
        last = mid - 1;
      } else {
        first = mid + 1;
      }
    }
  }
  return found;
};
var arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13];
console.log(binarySearch(arr, 11));
console.log(binarySearch(arr, 25));
console.log(binarySearch(arr, 2));
